eslint-rules/ts-no-use-before-define.js

```diff
@@ -0,0 +1,288 @@
+// Added here until https://github.com/nzakas/eslint-plugin-typescript/pull/34
```

Comment:
Though thinking about this more, this is a funny rule because I'd expect the TypeScript compiler itself to catch this. So I'm not sure why we need it 🤔